src/data/dynamodb.py

```diff
@@ -0,0 +1,291 @@
+
+    def update_tiles(self, tile_pool_id, removals=None, insertions=None):
+        print(
+            f"Updating tiles for pool ID: {tile_pool_id}, \
+                removals: {removals}, insertions: {insertions}"
+        )
+        if removals is None and insertions is None:
+            print("No changes to update")
+            return False
+
+        current_pool = self.get_tile_pool(tile_pool_id)
+        if current_pool is None:
+            print("Tile pool not found")
+            return False
+
+        try:
+            if removals:
+                old_pool = current_pool["tiles"]
+                new_tiles = frozenset(tile for tile in old_pool.tiles if tile.text not in removals)
+                self._update_tiles(tile_pool_id, new_tiles)
+                print("Removed tiles successfully")
+
+            if removals and insertions:
+                current_pool = self.get_tile_pool(tile_pool_id)
+                if current_pool is None:
+                    print("Tile pool not found")
+                    return False
+
+            if insertions:
+                old_pool = current_pool["tiles"]
+
+                # TODO: add free tile to insertions
+                new_pool = list(old_pool.tiles) + insertions
+                print(f"New pool: {new_pool}")
+                # new_pool = TilePool(frozenset(insertions))
+                # new_pool_tiles = list(new_pool.tiles)
+                # new_pool_tiles.append(old_pool)
+                # if new_pool.free:
+                #     new_pool_tiles.append(new_pool.free)
+
+                self._update_tiles(tile_pool_id, new_pool)
+                print("Inserted tiles successfully")
+
+            return True
+        except Exception as e:
+            print(f"Error updating tiles: {e}")
+            return False
+
+    def _update_tiles(self, tile_pool_id, new_pool):
+        print(f"Updating pool for ID: {tile_pool_id} with new pool data")
+
+        # Convert the frozenset of Tile objects to a list of dictionaries
+        new_tiles_list = [tile_to_dict(tile) for tile in new_pool]
+
+        # Use the utility function to convert to DynamoDB format
+        new_pool_dynamodb = self._dict_to_dynamodb({"new_pool": new_tiles_list})
+
+        try:
+            self.client.update_item(
+                TableName=self.table_name,
+                Key={"id": {"S": tile_pool_id}},
+                UpdateExpression="SET tiles = :new_pool",
+                ExpressionAttributeValues={":new_pool": new_pool_dynamodb["new_pool"]},
+            )
+            print("Update successful")
+            return True
+        except Exception as e:
+            print(f"Error in _update_tiles: {e}")
+            return False
+
+    def get_tile_pools(self, size=None, page=None, sort=SortMethod.DEFAULT, sort_asc=True):
+        print(
+            f"Getting tile pools, size: {size}, page: {page}, sort: {sort}, ascending: {sort_asc}"
+        )
+
+        if size is None or size < 1:
```

Comment:
It's ok to leave it as is, but the super has methods for paginating and sorting a list